### PR TITLE
Fixes spirit holding component not working on null rods by adding the subtype picker component to null rods

### LINKED ISSUE #60712: possessed blades cannot awaken a spirit
## Round ID: Tested locally on TG master

[Round ID]: # 

## Reproduction:

1) Take a possessed blade
2) Use it in hand
3) Notice it doesn't work at all

I've been trying to get this to work and it just doesn't, it started from I assume #60614 

Giving this component to any item (except the possessed blade) works, it only doesn't work specifically on the possessed blade (from my testing), and I don't know why.
I've even tried removing the component from the blade and giving it via VV, and that hasn't worked either.

From what I see, the posessed blade registers the signals, but using in hand, and only using in hand, is the only thing that doesn't work. If you try to use it in hand, it does literally nothing. Examining it does work however.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Null rods were not calling parent on attack_self and thus not sending signals to the component. DUMB! STINKY!!!!

I have elected to make subtype picking a component itself so this mistake will never be remade

fixes #60712 

also, fixes spirit holder not making sounds

## Why It's Good For The Game

fix! but also refactor!

## Changelog
:cl:
fix: spirit holder exorcisms now make sounds
fix: fixes possessed blade not actually being able to be possessed
refactor: null rod's subtype picking is now an element
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
@timothymtorres Hey why'd you remove the stale label here?

**Comment by timothymtorres** [comment]
> @timothymtorres Hey why'd you remove the stale label here?

Discord_NebQPYsvkn

**Comment by LemonInTheDark** [comment]
@tralezab 
if you want something destaled you should get a maint interested in it, otherwise you're just getting someone to press the reset button on waiting for you, which is silly. Stale exists to get people to sell maints on a pr.

File: code/game/objects/items/holy_weapons.dm [hunk 1]
```diff
@@ -462,8 +414,7 @@
 	hitsound = 'sound/weapons/chainsawhit.ogg'
 	tool_behaviour = TOOL_SAW
 	toolspeed = 0.5 //faster than normal saw
-	chaplain_spawnable = FALSE
```
Comment (Watermelon914) [inline code/game/objects/items/holy_weapons.dm:465]: I think you should keep the chaplain_spawnable variable. It's unclear that you have to set the menu_description to null to tell if the nullrod type is spawnable, even if it is in the documentation.

File: code/datums/components/subtype_picker.dm [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * subtype picker component allows for an item to transform into its subtypes (this is not enforced and you can turn in whatever types, but
+ * i used this name as it was incredibly accurate for current usage of the behavior)
+ *
+ * Used for the null rod to pick the other holy weapons.
+ */
+/datum/component/subtype_picker
+	///on Attach, it's a list of types and their menu descriptions. after building, it's a list of lists with all the data each item needs to give to the radial menu.
```
Comment (LemonInTheDark) [inline code/datums/components/subtype_picker.dm:8]: I really dislike this pattern. Wait you're not even modifying `subtype2descriptions` later what does this mean

File: code/datums/components/subtype_picker.dm [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	var/list/subtype2descriptions
+	///list given to the radial menu to display
+	var/list/built_radial_list
+	///the radial will return a name of the wanted subtype, this is a list of references back to the type
+	var/list/name2subtype
```
Comment (LemonInTheDark) [inline code/datums/components/subtype_picker.dm:13]: References is a weird way to describe this, makes me think it's an object

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Walter's review. Also is there any reason this needs to be a component? Admemery or something?

File: code/game/objects/items/holy_weapons.dm (hunk 1 above)
Comment (tralezab) [reply code/game/objects/items/holy_weapons.dm:465]: people should be reading the documentation though, if someone is wondering how weapons are made spawnable or not they'd simply read what i've docced, no?

File: code/datums/components/subtype_picker.dm (hunk 2 above)
Comment (tralezab) [reply code/datums/components/subtype_picker.dm:8]: incorrect doc, sorry. just having it not mutate makes this less dislikable, yes?

File: code/datums/components/subtype_picker.dm (hunk 3 above)
Comment (tralezab) [reply code/datums/components/subtype_picker.dm:13]: i think at one point when it was an element it may have been. either way incorrect doc

**Comment by tralezab** [comment]
>Also is there any reason this needs to be a component? Admemery or something?

We have item renaming and retheming at an atom level and that's typeless retheming. having a possible component to use in the future for typed retheming is super useful

also the item retheming should be a component

File: code/game/objects/items/holy_weapons.dm (hunk 1 above)
Comment (Watermelon914) [reply code/game/objects/items/holy_weapons.dm:465]: Variable names should usually be just as clear as the documentation.

File: code/datums/components/subtype_picker.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/datums/components/subtype_picker.dm]: yes very much. vars meaning one thing is more then enough

File: code/game/objects/items/holy_weapons.dm (hunk 1 above)
Comment (tralezab) [reply code/game/objects/items/holy_weapons.dm:465]: okay, i'll change it 👍

**Pull request merged**